### PR TITLE
[Merged by Bors] - feat(Data/*/Sort): lemmas on sorted lists

### PR DESCRIPTION
**Pull request by vihdzp**
The main thing we prove is that if `l` is a list, and `a` is smaller than all elements on it, then `sort r (a :: l) = a :: sort r l`.

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Data/Finset/Sort.lean [hunk 1]
```diff
@@ -64,11 +64,24 @@ theorem sort_empty : sort r ∅ = [] :=
 theorem sort_singleton (a : α) : sort r {a} = [a] :=
   Multiset.sort_singleton r a
 
+theorem sort_insert [DecidableEq α] {a : α} {s : Finset α} (h₁ : ∀ b ∈ s, r a b) (h₂ : a ∉ s) :
+    sort r (insert a s) = a :: sort r s := by
```
Comment (Ruben-VandeVelde) [inline Mathlib/Data/Finset/Sort.lean:68]: I wonder if a variant with `Finset.cons` would be useful, and whether you could reprove this one in terms of it

**Review by Ruben-VandeVelde — APPROVED** [review]

File: Mathlib/Data/Finset/Sort.lean (hunk 1 above)
Comment (vihdzp) [reply Mathlib/Data/Finset/Sort.lean:76]: Sure, I've added it to the PR.

**Review by Ruben-VandeVelde — APPROVED** [review]
maintainer merge

**Comment by jcommelin** [comment]
Thanks :tada:

bors merge

**Pull request closed without merging**